src/components/SelectState.js

```diff
@@ -1,23 +1,62 @@
 // @flow
-
 import React, { Component } from 'react';
 import wrapWithClickout from 'react-clickout';
-import SelectBase, { basePropTypes, baseDefaultProps } from './SelectBase';
-import type { SelectState, SelectBaseProps, SelectStateDefaultProps } from '../types';
 
-export class SelectStateComponent extends
-  Component<SelectStateDefaultProps, SelectBaseProps, SelectState> {
-  static defaultProps: SelectStateDefaultProps;
-  static props: SelectBaseProps;
+import SelectBase from './SelectBase';
+
+import type { Options, Selected, Callback, Styles, DefaultStyles } from '../types';
+
+type SelectStateProps = {
+  // config
+  id: string,
+  isMultipleSelect: boolean,
+  isSearchable: boolean,
+
+  // data / appearance
+  label: string,
+  placeholder: string,
+  options: Options,
+  styles: Styles,
+
+  // methods
+  onCheck: Callback,
 
-  constructor(props: SelectBaseProps) {
+  // dynamic
+  selected: Selected,
+  searchTerm: string,
+};
+
+type SelectStateDefaultProps = {
+  // config
+  isMultipleSelect: false,
+  isSearchable: false,
+
+  // data / appearance
+  label: '',
+  placeholder: '',
+  styles: DefaultStyles,
+
+  // dynamic
+  searchTerm: '',
+};
+
+type SelectStateState = {
+  isOpen: boolean,
+  options: Options,
+}
+
+export class SelectStateComponent extends Component {
+  defaultProps: SelectStateDefaultProps;
```

Comment:
we will discuss today. I was able to see type errors with generics, lets see whats going wrong